### PR TITLE
Store LiteralInteger value in a private variable

### PR DESCRIPTION
**Pull request by EmilyBourne**
**Commit Summary**
- Store LiteralInteger value in a private variable
- Remove dead code in parser/semantic.py
- Use operations rather than calling sympy operations directly

### CODE REVIEW

**Review by MenoIy — APPROVED** [review]

File: pyccel/ast/sympy_helper.py [hunk 1]
```diff
@@ -139,7 +142,7 @@ def pyccel_to_sympy(expr, symbol_map, used_names):
 
     elif isinstance(expr, PyccelMul):
         args = [pyccel_to_sympy(e, symbol_map, used_names) for e in expr.args]
-        return sp.Mul(*args)
+        return args[0] * args[1]
```
Comment (yguclu) [inline pyccel/ast/sympy_helper.py:145]: Is our `PyccelMul` a binary operation, or can it have more than two arguments? Sorry but I don't remember

File: pyccel/ast/sympy_helper.py [hunk 2]
```diff
@@ -151,11 +154,11 @@ def pyccel_to_sympy(expr, symbol_map, used_names):
 
     elif isinstance(expr, PyccelAdd):
         args = [pyccel_to_sympy(e, symbol_map, used_names) for e in expr.args]
-        return sp.Add(*args)
+        return args[0] + args[1]
```
Comment (yguclu) [inline pyccel/ast/sympy_helper.py:157]: Same question as for `PyccelMul`

File: pyccel/ast/sympy_helper.py (hunk 1 above)
Comment (EmilyBourne) [reply pyccel/ast/sympy_helper.py:145]: Yes, `PyccelMul` inherits from `PyccelBinaryOperator`. This mimicks the way python's ast handles this (arithmetic operators are all instances of `ast.BinOp`)

File: pyccel/ast/sympy_helper.py (hunk 2 above)
Comment (EmilyBourne) [reply pyccel/ast/sympy_helper.py:157]: Yes, see above

File: pyccel/ast/sympy_helper.py (hunk 1 above)
Comment (yguclu) [reply pyccel/ast/sympy_helper.py:145]: Thanks

**Review by yguclu — APPROVED** [review]

**Pull request merged**